src/Abstractions/BaseTypes.cs

```diff
@@ -131,4 +132,58 @@ public CharBoard Set(int r, int c, char value)
             return new CharBoard(Size, newCells);
         }
     }
+    
+    public record DiceBoard
+    {
+        private static readonly ConcurrentDictionary<int, DiceBoard> EmptyCache = new();
+        public static DiceBoard Empty(int size) => EmptyCache.GetOrAdd(size, size1 => new DiceBoard(size1));
+
+        public int Size { get; }
+        public Dictionary<int, string[]> Cells { get; }
+
+        public string[] this[int r, int c] {
+            get {
+                var cellIndex = GetCellIndex(r, c);
+                if (cellIndex < 0 || cellIndex >= Cells.Count)
+                    return new string[] {"lightblue", "lightblue", "lightblue", "lightblue"};
```

Comment:
IMO it's a bad idea to use string as underlying player color or something like this. Maybe an enum?